### PR TITLE
added msol unit test cases

### PR DESCRIPTION
**Pull request by snehalkotwal**
1. Set-EntraBetaDirSyncConfiguration cmdlet issue is solved and updated doc in the same PR.
2. All unit test cases are working.

### CODE REVIEW

File: test/module/Entra/Set-EntraDirSyncFeature.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        )
+        }
+    Mock -CommandName Get-MgDirectoryOnPremiseSynchronization -MockWith $scriptblock -ModuleName Microsoft.Graph.Entra
+    
+    Mock -CommandName Update-MgDirectoryOnPremiseSynchronization -MockWith {} -ModuleName Microsoft.Graph.Entra
+}
+Describe "Set-EntraDirSyncFeature" {
+    Context "Test for Set-EntraDirSyncFeature" {
+        It "Should sets identity synchronization features for a tenant." {
+            $result =  Set-EntraDirSyncFeature -Feature "BypassDirSyncOverrides" -Enable $false -TenantId "00aa00aa-bb11-cc22-dd33-44ee44ee44ee" -Force
```
Comment (KenitoInc) [inline test/module/Entra/Set-EntraDirSyncFeature.Tests.ps1:24]: Docs have `Enabled` as the parameter.
image
Ps1 file has `Enabled` https://github.com/microsoftgraph/entra-powershell/blob/2cfcd9a542909d717731c08f9ecd5afff02869e8/module/Entra/AdditionalFunctions/Set-EntraDirSyncFeature.ps1#L9

These tests have `Enable`. Should be fixed.

File: test/module/Entra/Set-EntraDomainFederationSettings.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                } 
+                "Parameters" = $args
+            }
+        )
+    }    
+    Mock -CommandName Get-MgDomainFederationConfiguration -MockWith $scriptblock -ModuleName Microsoft.Graph.Entra
+
+    Mock -CommandName Update-MgDomainFederationConfiguration -MockWith {} -ModuleName Microsoft.Graph.Entra
+}
+Describe "Set-EntraDomainFederationSettings" {
```
Comment (KenitoInc) [inline test/module/Entra/Set-EntraDomainFederationSettings.Tests.ps1:36]: Add tests that use these params
```
SigningCertificate
NextSigningCertificate
```

File: test/module/EntraBeta/Convert-EntraBetaFederatedUser.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            }        
+    )   
+ }  
+
+    Mock -CommandName Get-MgBetaUserAuthenticationMethod -MockWith $scriptblockForAuthenticationMethod -ModuleName Microsoft.Graph.Entra.Beta
+    Mock -CommandName Get-MgBetaUser -MockWith $scriptblockForMgUser -ModuleName Microsoft.Graph.Entra.Beta
+    Mock -CommandName Reset-MgBetaUserAuthenticationMethodPassword -MockWith {} -ModuleName Microsoft.Graph.Entra.Beta
+}
+ 
+    Describe "Convert-EntraBetaFederatedUser" {
```
Comment (KenitoInc) [inline test/module/EntraBeta/Convert-EntraBetaFederatedUser.Tests.ps1:29]: This cmdlet doesn't exist

File: test/module/EntraBeta/Get-EntraBetaDirSyncFeature.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                    "UserForcePasswordChangeOnLogonEnabled" = $false;
+                    "UserWritebackEnabled" = $false;
+                   }
+            }
+        )
+    }    
+    Mock -CommandName Get-MgBetaDirectoryOnPremiseSynchronization -MockWith $scriptblock -ModuleName Microsoft.Graph.Entra.Beta
+}
+
+Describe "Get-EntraDirSyncFeature" {
```
Comment (KenitoInc) [inline test/module/EntraBeta/Get-EntraBetaDirSyncFeature.Tests.ps1:40]:
```suggestion
Describe "Get-EntraBetaDirSyncFeature" {
```

File: test/module/EntraBeta/Get-EntraBetaDirSyncFeature.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    Context "Test for Get-EntraDirSyncFeature" {
```
Comment (KenitoInc) [inline test/module/EntraBeta/Get-EntraBetaDirSyncFeature.Tests.ps1:41]:
```suggestion
    Context "Test for Get-EntraBetaDirSyncFeature" {
```

File: test/module/EntraBeta/Get-EntraBetaFederationProperty.Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    Import-Module (Join-Path $psscriptroot "..\Common-Functions.ps1") -Force
+
+    $scriptblock = {
+        return @(
+            [PSCustomObject]@{
+                "ActiveSignInUri"     = "https://sts.anmaji.myworkspace.microsoft.com/adfs/services/trust/2005/usernamemixed"
+                "DisplayName"         = "ADFS HYPER-V LAB"
+                "IssuerUri"           = "http://anmaji.myworkspace.microsoft.com/adfs/services/trust/"
+                "MetadataExchangeUri" = "https://sts.anmaji.myworkspace.microsoft.com/adfs/services/trust/mex"  
+                "PassiveSignInUri"    = "https://sts.anmaji.myworkspace.microsoft.com/adfs/ls/"
```
Comment (KenitoInc) [inline test/module/EntraBeta/Get-EntraBetaFederationProperty.Tests.ps1:17]: Avoid `microsoft.com` domains

File: test/module/EntraBeta/Get-EntraBetaHasObjectsWithDirSyncProvisioningError.Tests.ps1 [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+BeforeAll {  
+    if((Get-Module -Name Microsoft.Graph.Entra.Beta) -eq $null){
+        Import-Module Microsoft.Graph.Entra.Beta      
+    }
+    Import-Module (Join-Path $psscriptroot "..\Common-Functions.ps1") -Force
+    
+    Mock -CommandName Invoke-GraphRequest -MockWith {} -ModuleName Microsoft.Graph.Entra.Beta
+}
+
+Describe "Get-EntraBetaHasObjectsWithDirSyncProvisioningError" {
```
Comment (KenitoInc) [inline test/module/EntraBeta/Get-EntraBetaHasObjectsWithDirSyncProvisioningError.Tests.ps1:13]: This cmdlet doesn't exist

File: test/module/EntraBeta/Get-EntraBetaPasswordPolicy.Tests.ps1 [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+            }
+        )
+    }    
+    Mock -CommandName Get-MgBetaDomain -MockWith $scriptblock -ModuleName Microsoft.Graph.Entra.Beta
+}
+
+Describe "Get-EntraBetaPasswordPolicy" {
+    Context "Test for Get-EntraBetaPasswordPolicy" {
+        It "Should gets the current password policy for a tenant or a domain." {
+            $result = Get-EntraBetaPasswordPolicy -DomainName "M365x99297270.onmicrosoft.com"
```
Comment (KenitoInc) [inline test/module/EntraBeta/Get-EntraBetaPasswordPolicy.Tests.ps1:26]:
```suggestion
            $result = Get-EntraBetaPasswordPolicy -DomainName "SawyerM@contoso.com"
```

File: test/module/EntraBeta/Get-EntraBetaPasswordPolicy.Tests.ps1 [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+            }
+        }
+        It "Should execute successfully without throwing an error" {
+            # Disable confirmation prompts       
+            $originalDebugPreference = $DebugPreference
+            $DebugPreference = 'Continue'
+    
+            try {
+                # Act & Assert: Ensure the function doesn't throw an exception
+                { Get-EntraBetaPasswordPolicy -DomainName "M365x99297270.onmicrosoft.com" -Debug } | Should -Not -Throw
```
Comment (KenitoInc) [inline test/module/EntraBeta/Get-EntraBetaPasswordPolicy.Tests.ps1:62]:
```suggestion
                { Get-EntraBetaPasswordPolicy -DomainName "SawyerM@contoso.com" -Debug } | Should -Not -Throw
```

**Review by KenitoInc — CHANGES_REQUESTED** [review]

File: test/module/Entra/Convert-EntraFederatedUser.Tests.ps1 [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+            }        
+    )   
+ }  
+
+    Mock -CommandName Get-MgUserAuthenticationMethod -MockWith $scriptblockForAuthenticationMethod -ModuleName Microsoft.Graph.Entra
+    Mock -CommandName Get-MgUser -MockWith $scriptblockForMgUser -ModuleName Microsoft.Graph.Entra
+    Mock -CommandName Reset-MgUserAuthenticationMethodPassword -MockWith {} -ModuleName Microsoft.Graph.Entra
+}
+ 
+    Describe "Convert-EntraFederatedUser" {
```
[the remark this answers is not in the document]
Comment (SteveMutungi254) [reply test/module/Entra/Convert-EntraFederatedUser.Tests.ps1:29]: Was renamed to `Update-EntraBetaUserFromFederated` and `Update-EntraUserFromFederated`

File: test/module/Entra/Get-EntraPasswordPolicy.Tests.ps1 [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+            }
+        )
+    }    
+    Mock -CommandName Get-MgDomain -MockWith $scriptblock -ModuleName Microsoft.Graph.Entra
+}
+
+Describe "Get-EntraPasswordPolicy" {
+    Context "Test for Get-EntraPasswordPolicy" {
+        It "Should gets the current password policy for a tenant or a domain." {
+            $result = Get-EntraPasswordPolicy -DomainName "M365x99297270.onmicrosoft.com"
```
[the remark this answers is not in the document]
Comment (SteveMutungi254) [reply test/module/Entra/Get-EntraPasswordPolicy.Tests.ps1:26]: @KenitoInc : SawyerM@contoso.com is a UPN. The domain should be `contoso.com`.

File: test/module/Entra/Get-EntraPasswordPolicy.Tests.ps1 [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+BeforeAll {  
+    if((Get-Module -Name Microsoft.Graph.Entra) -eq $null){
+        Import-Module Microsoft.Graph.Entra        
+    }
+    Import-Module (Join-Path $psscriptroot "..\Common-Functions.ps1") -Force
+
+    $scriptblock = {
+        return @(
+            [PSCustomObject]@{
+                "Id" = "M365x99297270.onmicrosoft.com"
```
[the remark this answers is not in the document]
Comment (SteveMutungi254) [reply test/module/Entra/Get-EntraPasswordPolicy.Tests.ps1:13]: We should use contoso.com instead.

**Review by KenitoInc — APPROVED** [review]

**Pull request merged**